### PR TITLE
Fix CI on Julia non-master

### PR DESCRIPTION
**Pull request by jpsamaroo**
Julia master changed the field `min_world` to `primary_world` in `Core.Method`, so now we handle both cases.

### CODE REVIEW

**Comment by jpsamaroo** [comment]
I'm going to try to fix v1.0 by rebuilding the CI container, but if that fails, I'm willing to keep ignoring it for now :smile:

**Comment by jpsamaroo** [comment]
I see, we're not failing to find `ld.lld`, but apparently we can't set an `IOBuffer` as the `stdout` argument to `pipeline` on Julia <1.1 (and any error manifests as skipping that `ld.lld` path). I've just pushed a fix to HSARuntime.

File: src/execution.jl [hunk 1]
```diff
@@ -66,7 +66,11 @@ world_age() = ccall(:jl_get_tls_world_age, UInt, ())
 # slow lookup of local method age
 function method_age(f, tt)::UInt
     for m in Base._methods(f, tt, 1, typemax(UInt))
-        return m[3].min_world
+        @static if :min_world in fieldnames(Core.Method)
```
Comment (vchuravy) [inline src/execution.jl]: normally a version check is preferred.

Comment (jpsamaroo) [reply src/execution.jl]: Gotcha, I was just trying to be as lazy as possible here. Will figure out how to do that!

Comment (jpsamaroo) [reply src/execution.jl]: When it doubt, check what CUDAnative does :)

**Pull request merged**